### PR TITLE
Cherrypick doxygen

### PR DESCRIPTION
**Pull request by adramaxxx**
I have now cherry-picked the commits i wanted and left out the kahansummation. The code compiles, all tests are successful, and all the documentation is present

### CODE REVIEW

File: html/annotated.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
Comment (RasmusRendal) [inline html/annotated.html]: You should remove the directiories html and latex from your git index

**Review by RasmusRendal — CHANGES_REQUESTED** [review]

File: html/annotated.html (hunk 1 above)
Comment (adramaxxx) [reply html/annotated.html]: i see, although i put them in the .gitignore before i even made a first commit. Welp, no helping that

File: src/eulerevaluator.cpp [hunk 2]
```diff
@@ -38,6 +38,8 @@ void EulerEvaluator::InitializeEquationParts() {
 	}
 }
 
+//! \file
```
Comment (RasmusRendal) [inline src/eulerevaluator.cpp:41]: What's this for?

File: Doxyfile [hunk 3]
```diff
@@ -0,0 +1,2507 @@
...
+                         *.cxx \
+                         *.cpp \
+                         *.c++ \
+                         *.h \
+
+# The RECURSIVE tag can be used to specify whether or not subdirectories should
+# be searched for input files as well.
+# The default value is: NO.
+
+RECURSIVE              = NO
```
Comment (RasmusRendal) [inline Doxyfile]: This should probably be enabled so we can have `src/CRN` and such in the documentation

File: Doxyfile [hunk 4]
```diff
@@ -0,0 +1,2507 @@
...
+# This tag requires that the tag FILTER_SOURCE_FILES is set to YES.
+
+FILTER_SOURCE_PATTERNS =
+
+# If the USE_MDFILE_AS_MAINPAGE tag refers to the name of a markdown file that
+# is part of the input, its contents will be placed on the main page
+# (index.html). This can be useful if you have a project on for instance GitHub
+# and want to reuse the introduction page also for the doxygen output.
+
+USE_MDFILE_AS_MAINPAGE =
```
Comment (RasmusRendal) [inline Doxyfile]: Maybe we should set this to `README.md` so we don't have to repeat ourselves?

**Review by RasmusRendal — CHANGES_REQUESTED** [review]

File: src/eulerevaluator.cpp (hunk 2 above)
Comment (adramaxxx) [reply src/eulerevaluator.cpp:41]: a doxygen marker for it to scan the .cpp file, so i can include the snippet

File: Doxyfile (hunk 3 above)
Comment (adramaxxx) [reply Doxyfile]: alrighty then

File: Doxyfile (hunk 4 above)
Comment (adramaxxx) [reply Doxyfile]: good idea

File: src/eulerevaluator.cpp (hunk 2 above)
Comment (RasmusRendal) [reply src/eulerevaluator.cpp:41]: That's pretty neat

**Review by RasmusRendal — APPROVED** [review]

**Pull request merged**